pkg/gits/operations/pull_request_op.go

```diff
@@ -92,9 +92,11 @@ func (o *PullRequestOperation) CreatePullRequest(kind string, update ChangeFiles
 		if err != nil {
 			return nil, errors.Wrapf(err, "failed to create PR for base %s and head branch %s from temp dir %s", o.Base, details.BranchName, dir)
 		}
-		err = gits.AddLabelsToPullRequest(result, labels)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to add labels %+v to PR %s", labels, result.PullRequest.URL)
+		if result != nil {
```

Comment:
So what are we saying here? That `gits.PushRepoAndCreatePullRequest` might return nil and no error? Is there then not something semantically wrong in that function? In which cases do we get `nil` and no error?

Reply:
I think is this case inside of the function:
```
		changed, err := gitter.HasChanges(dir)
		if err != nil {
			return nil, errors.WithStack(err)
		}
		if !changed {
			log.Logger().Warnf("No changes made to the source code in %s. Code must be up to date!", dir)
			return nil, nil
		}
```

Maybe it  makes sense to explicitly check if there are any changes before calling the function and skip everything afterwords. The logic would look more clear to me. This doesn't look like an error case.

Reply:
@hferentschik We get `nil` and no error if there are no changes to create a PR with. And @ccojocar, which function do you mean?

Reply:
`changed, err := gitter.HasChanges(dir)`

Reply:
I'll tackle that in a followup - I'm gonna rework this like #6244 to apply the labels at creation time instead of immediately after to deal with the nil pointer specifically.